File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void right(vector <int>&ans,int level,TreeNode* root){
        if(root==NULL){
            return;
        }
        if(level==ans.size())
            ans.push_back(root->val);
        right(ans,level+1,root->right);
        right(ans,level+1,root->left);
    }
    vector<int> rightSideView(TreeNode* root) {
        vector <int> ans;
        right(ans,0,root);
        return ans;
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> l = new ArrayList<>();
        reversePreOrder(root, 0, l);
        return l;
    }

    public void reversePreOrder(TreeNode root, int level, List<Integer> l) {
        if (root == null) {
            return;
        }
        if (l.size() == level)
            l.add(root.val);
        reversePreOrder(root.right, level + 1, l);
        reversePreOrder(root.left, level + 1, l);
    }
}